### PR TITLE
In_preparation and disabled organizations

### PR DESCRIPTION
**Pull request by fedescarpa**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -73,6 +75,14 @@ def validate_user_profile!
     end
   end
 
+  def validate_archived_organization!
+    raise Mumuki::Domain::ArchivedOrganizationError if Organization.current.archived? && !current_user.teacher?
+  end
+
+  def validate_enabled_organization!
+    raise Mumuki::Domain::DisabledOrganizationError if Organization.current.disabled? && !current_user.teacher?
```
Comment (flbulgarelli) [inline app/controllers/application_controller.rb:83]: Se are checking domain conditions, using domain objects and raising domain exceptions. Shouldn't we move this logic to domain, too?

Something like organization.validate_accessible_for(user)

I Even think there is already a method like that somewhere

Comment (luchotc) [reply app/controllers/application_controller.rb:83]: Yep, I agree.
I think that moving this logic to `validate_accessible!(item)` makes a lot of sense.

**Comment by luchotc** [comment]
@flbulgarelli Same here :)

**Comment by luchotc** [comment]
I'll add some flow tests before integrating this one, but we can still use a review @flbulgarelli

**Pull request merged**